### ISSUE TITLE
A setting and/or flag to add short version numbers with `cargo add`

### ISSUE THREAD

**Issue by docwilco** [rust-lang/cargo#12072]
### Problem

Currently, `cargo add` will add full latest version number to `Cargo.toml`. Especially in new projects (where a lot of `add` actions happen,) this might not be ideal. I at least, find myself removing patch and ofteneven minor version numbers, so that I can occasionally just remove `Cargo.lock` and get everything upgraded.

### Proposed Solution

```
[add]
version-specificity="major"
```
resulting in `cargo add tokio` adding
```
[dependencies]
tokio = "1"
```

and
```
[add]
version-specificity="minor"
```
resulting in `cargo add tokio` adding
```
[dependencies]
tokio = "1.28"
```

Possible settings `major|minor|patch` with `patch` being the default.

Not quite sure about commandline flags, am absolutely open to suggestions there.

### Notes

I am absolutely open to writing a PR for this.

**Comment by epage** [comment]
Before anything else I want to check in on the motivation

> Currently, cargo add will add full latest version number to Cargo.toml. Especially in new projects (where a lot of add actions happen,) this might not be ideal. I at least, find myself removing patch and ofteneven minor version numbers, so that I can occasionally just remove Cargo.lock and get everything upgraded.

Why do you need to change your version requirement to upgrade?  The version requirement is saying "at least `x.y.z` but not more than `x+1.0.0`.  `cargo update` and `rm Cargo.lock` will both allow updating any of your dependencies, independent of the version requirement.

**Comment by epage** [comment]
For some background, when we merged `cargo add` (see #10472), one of the concerns was

> 2. This is a high UX feature that will draw a lot of attention (ie Issue influx)
> 
> e.g.
> - killercup/cargo-edit#521
> - killercup/cargo-edit#126
> - killercup/cargo-edit#217

*Note:* killercup/cargo-edit#126 is very similar to your request

To reduce this, we explicitly decided:

> We've tried to reduce the UX influx by focusing the scope to preserving semantic information (custom sort order, comments, etc) but being opinionated on syntax (style of strings, etc)

**Comment by docwilco** [comment]
Aha, OK, I totally missed that bit. My apologies for wasting your time!
